### DISCUSSION TITLE
Vercel system vars in NextJS frontend

### DISCUSSION THREAD

**Discussion by ScreamZ** [vercel/next.js#18945] [Help]
Hello, 

I've the following code and I would like to enable analytics only when deployed by vercel in production. For that I used `VERCEL_GITHUB_DEPLOYMENT` and activated it only in production configuration in vercel.

```ts
const isProduction = typeof process.env.VERCEL_GITHUB_COMMIT_REF !== "undefined";
if (!firebase.apps.length && typeof window !== "undefined") {
  firebase.initializeApp(firebaseConfig);
  firebase.analytics().setAnalyticsCollectionEnabled(isProduction);
}
```

The thing is that variable is not available to the frontend and as you can see I use `typeof window !== "undefined"`.
I could use NEXT_PUBLIC_ but not sure how and about the returned values.

Someone told me to use a .env.production with `NEXT_PUBLIC_VERCEL_GITHUB_DEPLOYMENT=${VERCEL_GITHUB_DEPLOYMENT}`

Do you have any idea? Is still `typeof process.env.VERCEL_GITHUB_COMMIT_REF !== "undefined"` relevant or will I end with an empty string if not defined ?

**Comment by jozsefsallai** [comment]
Hello.

I'm pretty sure `process.env.NODE_ENV` works on the frontend too, so why don't you just do something like this?

```js
const isProduction = process.env.NODE_ENV === 'production';
if (!firebase.apps.length && typeof window !== "undefined") {
  firebase.initializeApp(firebaseConfig);
  firebase.analytics().setAnalyticsCollectionEnabled(isProduction);
}
```

Vercel will set `NODE_ENV` to `production` on ... well ... production builds, so this would work for what you want to achieve (and I think it's a bit cleaner too).
